Dockerfile

```diff
@@ -60,7 +60,7 @@ RUN bazel version
 RUN ${PIP} install --no-cache-dir --upgrade \
     absl-py~=0.11.0 \
     attrs~=19.3.0 \
-    cachetools~=3.1.1 \
+    cachetools>=3.1.1,<5.0 \
```

Comment:
I think I would prefer to update this dependency to: `cachetools~=4.2.0`. This way we can avoid matches across versions with potentially breaking changes.

Reply:
Sure? I'm just following `google-auth`'s lead which seems to be confident the API is stable enough from 2 until 5 (excluding)

Reply:
Yeah, whether or not `google-auth` "knows" that they do not use API that might have broken between those major versions is not truly my concern. Though I am slightly interested how any released package would be stable with such a range.

For the most part my concern is with our direct dependency not our indirect dependency... just because `google-auth` are using an API that does not break, does not mean that TFF is not currently or in the future. I would prefer to have a dependency install failure/warning vs. a runtime failure.

Reply:
Done and rebased